### ISSUE TITLE
Random KeyErrors when performing GET requests (possible race condition)

### ISSUE THREAD

**Issue by KyleKreutzer** [encode/httpx#581]
When performing a get request using httpx.client.get, I am (with every 1/10 requests or so) getting random key errors. The only argument being handed to this call is a 5 second timeout, a google "NID" cookie, and the URL. I've attached a text file with the error.
[KeyError.txt](https://github.com/encode/httpx/files/3909364/KeyError.txt)

**Comment by tomchristie** [comment]
Thanks for reporting this.

What's the simplest possible example that reproduces this reliably?
Is this only occuring against a specific URL, or can you determine various cases where its failing?

**Comment by KyleKreutzer** [comment]
I've only tried it against the Google Trends API (https://trends.google.com/trends/api/*). The simplest way to reproduce the error would be to pull [](https://github.com/KyleKreutzer/pytrends-async) and run the unit tests. The problem is occurring intermittently so it may take more than 1 run to have the issue appear, but any of those tests seem to reliably reproduce the problem.

**Comment by KyleKreutzer** [comment]
`        for i in range(100):
            async with httpx.Client() as c:
                resp = await c.get('https://trends.google.com/?geo={geo}'.format(geo='US'), timeout=5)
                time.sleep(1)`

This will produce the bug

**Comment by millefalcon** [comment]
Is this still reproducible ?
I couldn't reproduce it with `python3.6.9`,

**Comment by millefalcon** [comment]
Is this still reproducible ?
Couldn't reproduce it with:
```
python3.6.9
trio==0.13.0
git+https://github.com/encode/httpx@75a439d7e2120edcddd8213df27dfeea28a2dc8c#egg=httpx
```

**Comment by KyleKreutzer** [comment]
I was able to reproduce it again with above code snippet on both mac and windows using 3.7.0 and 3.7.3. I am not using trio and am using version httpx==0.8.0 from pypi.

**Comment by tomchristie** [comment]
Closed in master via #612
